### ISSUE TITLE
Repo manager leaves files on fail

### ISSUE THREAD

**Issue by david4096** [ga4gh/server#1052]
When attempting to add the apparently corrupted file `HG00101.mapped.ILLUMINA.bwa.GBR.low_coverage.20130415.bam` I found the repo manager left behind the linked file, even though the process apparently failed.

```
(server-1)➜  server-1 git:(master) ✗ python repo_dev.py add-readgroupset ga4gh-example-data 1kg-p3-subset ../1000genomes/HG00101.mapped.ILLUMINA.bwa.GBR.low_coverage.20130415.bam 
[W::bam_hdr_read] EOF marker is absent. The input is probably truncated.
Traceback (most recent call last):
  File "repo_dev.py", line 7, in <module>
    ga4gh.cli.repo_main()
  File "/Users/david/server-1/ga4gh/cli.py", line 1672, in repo_main
    runner.run()
  File "/Users/david/server-1/ga4gh/cli.py", line 1468, in run
    self.datasetName, self.filePath, self.moveMode)
  File "/Users/david/server-1/ga4gh/repo_manager.py", line 427, in addReadGroupSet
    pysam.index(destPath.encode('utf-8'))
  File "/Users/david/server-1/server-1/lib/python2.7/site-packages/pysam/utils.py", line 65, in __call__
    "\n".join(stderr)))
pysam.utils.SamtoolsError: 'samtools returned with error 1: stdout=, stderr=samtools index: "ga4gh-example-data/datasets/1kg-p3-subset/reads/HG00101.mapped.ILLUMINA.bwa.GBR.low_coverage.20130415.bam" is corrupted or unsorted\n'

(server-1)➜  server-1 git:(master) ✗ ls ga4gh-example-data/datasets/1kg-p3-subset/reads/HG00101.mapped.ILLUMINA.bwa.GBR.low_coverage.20130415.bam
ga4gh-example-data/datasets/1kg-p3-subset/reads/HG00101.mapped.ILLUMINA.bwa.GBR.low_coverage.20130415.bam
```

**Comment by dcolligan** [comment]
So in the case (like here) where we are doing a processing step that may fail, what do you propose the recovery process should be?  I think:

0) delete all index files that pysam may leave lying around, in any case
1a) if the moveMode is link, delete the link
1b) if the moveMode is copy, delete the file
1c) if the moveMode is move, move the file back to where it came from

Sound good?

**Comment by david4096** [comment]
I believe if you were to perform the move after all other processing you would simplify the approach. If the index step's exceptions are handled properly you shouldn't have to do anything else.

**Comment by dcolligan** [comment]
Perhaps, but I want to avoid modifying files in areas that we don't control (e.g. outside a repo) and aren't explicitly asked to do

**Comment by david4096** [comment]
How about #1064? We should limit the features of the repo manager as much as possible and add features if their requested.

**Comment by jeromekelleher** [comment]
I'm with @david4096 here. Let's pull back on the auto-generating index feature of repo manager for a while, and flesh out the features we need for the upcoming milestones. (We'll figure out exactly what those features are over the next few days.)

**Comment by jeromekelleher** [comment]
Closed by #1166.
